### PR TITLE
Fix nxos_file_copy option value path validation

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

*  Modify `local_file`, `local_file_directory` and
   `remote_file` option type from `str` to `path`
   so that the option value is validated in Ansible
   for a legitimate path value

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

nxos_file_copy
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/plugins/action/nxos_file_copy.py [hunk 1]
```diff
@@ -74,22 +75,22 @@ def process_playbook_values(self):
...
+                elif option_type == 'bool':
+                    playvals[key] = validation.check_type_bool(playvals[key])
+                elif option_type == 'path':
+                    playvals[key] = validation.check_type_path(playvals[key])
+                else:
+                    raise AnsibleError('Unrecognized type <{0}> for playbook parameter <{1}>'.format(type, key))
 
-            if not type_ok:
-                raise AnsibleError('Playbook parameter <{0}> value should be of type <{1}>'.format(key, type))
+            except (TypeError, ValueError) as e:
```
Comment (mikewiebe) [inline lib/ansible/plugins/action/nxos_file_copy.py:90]: @ganeshrn The tests in https://github.com/ansible/ansible/blob/devel/test/integration/targets/nxos_file_copy/tests/cli/input_validation.yaml need to be updated to handle the new error message.

Also, I tried running the test with this code bug I see the following error message:

```
The full traceback is:
Traceback (most recent call last):
  File "/root/agents-ci/ansible/lib/ansible/executor/task_executor.py", line 146, in run
    res = self._execute()
  File "/root/agents-ci/ansible/lib/ansible/executor/task_executor.py", line 645, in _execute
    result = self._handler.run(task_vars=variables)
  File "/root/agents-ci/ansible/lib/ansible/plugins/action/nxos_file_copy.py", line 499, in run
    self.file_push()
  File "/root/agents-ci/ansible/lib/ansible/plugins/action/nxos_file_copy.py", line 212, in file_push
    if not self.local_file_exists(local_file):
  File "/root/agents-ci/ansible/lib/ansible/plugins/action/nxos_file_copy.py", line 165, in local_file_exists
    return os.path.isfile(file)
  File "/usr/lib/python2.7/genericpath.py", line 37, in isfile
    st = os.stat(path)
TypeError: coercing to Unicode: need string or buffer, NoneType found
```

It does not appear to be generating the failure message `argument ... is of type..`

**Review by mikewiebe — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/action/nxos_file_copy.py (hunk 1 above)
Comment (ganeshrn) [reply lib/ansible/plugins/action/nxos_file_copy.py:90]: Fixed review comments

**Comment by mikewiebe** [comment]
@ganeshrn I ran the tests again with the latest diffs but they are still failing.  I suspect the tests will also fail if you run them against your n9kv?  Can you confirm?

**Comment by ganeshrn** [comment]
@mikewiebe Did you update the integration test file as well from the PR? I have updated the test cases to reflect new changes.

**Comment by mikewiebe** [comment]
> @mikewiebe Did you update the integration test file as well from the PR? I have updated the test cases to reflect new changes.

Yes, I did get the latest updates.  Do they all pass on your testbed?

**Comment by ganeshrn** [comment]
@mikewiebe Based on the error message you shared it seems that your environment is still running an old test case. I have replaced the error message you are getting with https://github.com/ansible/ansible/blob/1d4fdbcfa1238e5200e19f964a01a9d8a1a837c1/test/integration/targets/nxos_file_copy/tests/cli/input_validation.yaml#L13 in this PR 

Also, the failing assert statement `result is search('Playbook parameter <remote_file> value should be of type <str>')` that you shared is not present in the updated integration test file.

**Comment by mikewiebe** [comment]
> @mikewiebe Based on the error message you shared it seems that your environment is still running an old test case. I have replaced the error message you are getting with
> 
> https://github.com/ansible/ansible/blob/1d4fdbcfa1238e5200e19f964a01a9d8a1a837c1/test/integration/targets/nxos_file_copy/tests/cli/input_validation.yaml#L13
> 
> in this PR
> Also, the failing assert statement `result is search('Playbook parameter <remote_file> value should be of type <str>')` that you shared is not present in the updated integration test file.

@ganeshrn Your right, my mistake.  I must have run the test from the wrong environment.  Looks good.

👍

**Review by mikewiebe — APPROVED** [review]

**Pull request merged**